src/dbnode/storage/block/wired_list.go

```diff
@@ -204,7 +204,12 @@ func (l *WiredList) Stop() error {
 //
 // We use a channel and a background processing goroutine to reduce blocking / lock contention.
 func (l *WiredList) BlockingUpdate(v DatabaseBlock) {
+	l.RLock()
+	if l.updatesCh == nil {
+		return
```

Comment:
Need RUnlock() before return here.

Reply:
need equivalent in NonBlockingUpdate, too

Reply:
Ah yeah, thought the `select` handled that case, but seems like that's incorrect